src/lib/hooks/routing/useRoutingAPIArguments.ts

```diff
@@ -0,0 +1,40 @@
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { useMemo } from 'react'
+
+/**
+ * Returns query arguments for the Routing API query or undefined if the
+ * query should be skipped.
```

Comment:
Include in the doc comment that the input does not need to be memoized, as it will be destructured.